src/MongoDB.Bson/Serialization/Conventions/ImmutableTypeClassMapConvention.cs

```diff
@@ -25,28 +25,34 @@ namespace MongoDB.Bson.Serialization.Conventions
     /// </summary>
     public class ImmutableTypeClassMapConvention : ConventionBase, IClassMapConvention
     {
+        private readonly BindingFlags _constructorBindingFlags;
+        private readonly BindingFlags _propertyBindingFlags;
+
+        /// <inheritdoc />
+        public ImmutableTypeClassMapConvention()
+        {
+            _constructorBindingFlags = BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic;
+            _propertyBindingFlags = BindingFlags.Instance | BindingFlags.Public;
```

Comment:
The two fields are only used in one place.

Can you make them local variables (before lines 48 and 55) instead?

Reply:
done